Allow hiding test state for first-contentful-paint tests

sefeng211 opened the pull request:
The testharness' `show_status` function call interfere
the tests result because it would display a text to the document
while the tests are running.

The issue is this text could be considered as the first-contentful-paint,
which is not what we expect.

We fix it by adding a new setup property to not displaying this text.

This PR intends to fix #24089

sefeng211 commented:
cc @normr @npm1

npm1 reviewed (commented):
paint-timing change looks fine to me, although perhaps we want to apply this to all paint-timing tests and not just those using test_fcp?

sefeng211 committed:
Allow hiding test state in testharness

Some tests are sensitive to the content of the document, thus
show_status may interfere the test results.

We add a new setup property to allow hiding the test state output.

sefeng211 committed:
Add hide_test_state property to the document

sefeng211 committed:
Hide test state for all paint timing tests

sefeng211 commented:
@npm1 Agreed. 

PR updated.

npm1 commented on resources/testharness.js:
```diff
@@ -2510,6 +2510,7 @@ policies and contribution forms [3].
         this.test_done_callbacks = [];
         this.all_done_callbacks = [];
 
+        this.do_notify_test_state = true;
```
Oh sorry I missed this before you added more but it probably makes sense to make this `hide_test_state = false` since we generally want the default value for booleans to be false.

npm1 reviewed (commented):
Thanks! I'm fine with landing this % comment, but have someone from infra take a look?

npm1 reviewed (approved):
paint-timing LGTM

Hexcles commented:
This seems very reasonable to me, but I'm not sure if we should do an RFC @web-platform-tests/wpt-core-team .

Ms2ger commented:
We probably should do an rfc, yeah

sefeng211 commented:
Not sure what an rfc is, is there any action that I should take?

Ms2ger commented:
See https://github.com/web-platform-tests/rfcs

sefeng211 commented:
The RFC is merged. 

I also updated the patch to follow @stephenmcgruer's [suggestion](https://github.com/web-platform-tests/rfcs/pull/58#pullrequestreview-439456850), which makes sense because `show_status` only outputs the test state while tests are running.

Hexcles reviewed (approved):
LGTM.

Thanks for following through the process, Sean!

stephenmcgruer commented:
Failed `chrome-dev-stability`:

```
4:15.46 INFO ## Unstable results ##
 4:15.46 INFO |                    Test                    |                  Subtest                   |          Results           |                                     Messages                                    |
 4:15.46 INFO |--------------------------------------------|--------------------------------------------|----------------------------|---------------------------------------------------------------------------------|
 4:15.46 INFO | `/paint-timing/fcp-only/fcp-gradient.html` | `Gradients should not count as contentful` | **FAIL: 9/10, PASS: 1/10** | `assert_equals: First contentful paint marked too early.  expected 0 but got 1` |
 4:15.46 INFO 
```

From wpt.fyi history, looks like a previously existing flake but I'm a little interested here just because the goal of this PR was to remove unexpectedly-early FCPs (due to test infra involvement). 

@sefeng211 - I leave this up to you; should this PR be admin-merged, or is this flake something that is worth pausing to investigate?

npm1 commented:
I think the flake there is caused by Chrome's failure to implement FCP properly for that particular testcase. If it's flaky it probably means that the timeout may not be long enough to guarantee that the incorrectly early FCP is always fired, but this is unrelated to this PR.

stephenmcgruer commented:
Ack, admin-merging. :)

Hexcles commented:
If this is a Chromium bug, it'd be great to file an issue on crbug.

npm1 commented:
There's already https://bugs.chromium.org/p/chromium/issues/detail?id=1062984 (apparently flakiness is not fixed though?) and https://bugs.chromium.org/p/chromium/issues/detail?id=1071450
